### ISSUE TITLE
📖 [Story] Improve the `torch_tensorrt.dynamo.compile` path

### ISSUE THREAD

**Issue by gs-olive** [pytorch/TensorRT#1941]
## TL;DR
Improve the performance acceleration, robustness, and overall quality of the `torch_tensorrt.dynamo.compile` path. Simultaneously use these improvements to improve the quality of the `torch_tensorrt.dynamo.fx_ts_compat` path.

## Goal(s)
By adding lowering passes, improving converter coverage, and building features to accelerate the performance of our `torch.compile` backend, we can improve the quality of the path. Additionally, in conjunction with #1940, many of these upgrades can be ported over to our `torch._dynamo.export` path.

## Implementation Phases
1. Module-Level Acceleration #1894 
2. Improved Converter Coverage (#1769, #1754)
3. Support for `truncate_long_and_double` (#1865)
4. Engine caching + cross-session state capture
5. Dynamic shape support + avoiding unnecessary recompilation

## Additional context
Related to #1940
